### ISSUE TITLE
Handle panic in user function in Go

### ISSUE THREAD

**Issue by matejvasek** [boson-project/buildpacks#23]
Probably not necessary, but I think it might be useful.

For HTTP it could look like:
```go
func makeSafeHandler(handler http.Handler) http.Handler {
	f := func(rw http.ResponseWriter, req *http.Request) {
		defer func() {
			if r := recover(); r != nil {
				rw.WriteHeader(http.StatusInternalServerError)
				rw.Header().Add("Content-Type", "text/plain")
				fmt.Fprintf(rw, "user function error: %v\n", r)
			}
		}()
		handler.ServeHTTP(rw, req)
		return
	}
	return http.HandlerFunc(f)
}
```

**Comment by lance** [comment]
Seems useful for debugging, I agree.

**Comment by matejvasek** [comment]
@lkingland wdyt?

**Comment by matejvasek** [comment]
related:
https://github.com/cloudevents/sdk-go/issues/584
https://github.com/cloudevents/sdk-go/pull/585

**Comment by matejvasek** [comment]
this issue could be closed once we will use `sdk-go v2.3`

**Comment by matejvasek** [comment]
I updated `sdk-go` version in our templates in `faas` repository.
